src/lib/array.jl

```diff
@@ -957,3 +959,21 @@ end
   back(Δ::AbstractArray) = (nothing, getindex.(_back.(Δ), 1))
   return Fill(y, size(r)), back
 end
+
+# Sparse Arrays
+
+@adjoint function SparseMatrixCSC{T,N}(arr) where {T,N}
+  SparseMatrixCSC{T,N}(arr), Δ -> (collect(Δ),)
+end
+
+@adjoint function SparseVector{T,N}(v) where {T,N}
+  SparseVector{T,N}(v), Δ -> (collect(Δ),)
+end
+
+@adjoint diagm(x::AbstractSparseArray) = diagm(x), Δ -> (diag(Δ), )
+
+@adjoint function Broadcast.broadcasted(Float32, a::AbstractSparseArray{T,N}) where {T,N}
```

Comment:
```suggestion
@adjoint function Broadcast.broadcasted(::Type{Float32}, a::AbstractSparseArray{T,N}) where {T,N}
```

otherwise it's just going to overload every case!